File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/cashin/ui/CashInPresenterModule.java
package com.fintechplatform.ui.cashin.ui;

import com.fintechplatform.api.card.api.PaymentCardAPI;
import com.fintechplatform.api.cashin.api.CashInAPI;
import com.fintechplatform.ui.account.balance.helpers.BalanceHelper;
import com.fintechplatform.ui.card.db.PaymentCardPersistenceDB;
import com.fintechplatform.ui.cashin.CashInContract;
import com.fintechplatform.ui.cashin.TopUpPresenter;
import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.ui.money.FeeHelper;
import com.fintechplatform.ui.money.MoneyHelper;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class CashInPresenterModule {

    private final CashInContract.View view;
    private final DataAccount configuration;

    public CashInPresenterModule(CashInContract.View view,
                                 DataAccount configuration) {
        this.view = view;
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    CashInContract.Presenter provideCashInPresenter(CashInAPI api,
                                                   PaymentCardAPI apiCard,
                                                   MoneyHelper moneyHelper,
                                                   BalanceHelper balanceHelper,
                                                   FeeHelper feeHelper,
                                                   PaymentCardPersistenceDB persistenceDB) {
        return new TopUpPresenter(configuration, view, api, apiCard, moneyHelper, balanceHelper, feeHelper, persistenceDB);
    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/cashin/di/CashInPresenterModule.kt
package com.fintechplatform.ui.cashin

import com.fintechplatform.api.card.api.PaymentCardAPI
import com.fintechplatform.api.cashin.api.CashInAPI
import com.fintechplatform.ui.account.balance.helpers.BalanceHelper
import com.fintechplatform.ui.card.db.PaymentCardPersistenceDB
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.money.FeeHelper
import com.fintechplatform.ui.money.MoneyHelper
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class CashInPresenterModule(private val view: CashInContract.View,
                            private val configuration: DataAccount) {

    @Provides
    @Singleton
    internal fun provideCashInPresenter(api: CashInAPI,
                                        apiCard: PaymentCardAPI,
                                        moneyHelper: MoneyHelper,
                                        balanceHelper: BalanceHelper,
                                        feeHelper: FeeHelper,
                                        persistenceDB: PaymentCardPersistenceDB): CashInContract.Presenter =
            CashInPresenter(configuration, view, api, apiCard, moneyHelper, balanceHelper, feeHelper, persistenceDB)
}